### PR TITLE
feat: Send flow UI updates

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
This PR aims to update the send flow UI on this figma [file](https://www.figma.com/file/mW1YrVHW1Ys7Q7d1MEDMPA/Send-fix---Remove-%22Recent%22-and-expand-%22transfer-between-my-accounts%22?type=design&node-id=109-2481&mode=design&t=QjnIhXor0Dc9dvRJ-0) 

Also, it was updated the Contacts view on settings was since they shared the same component

This PR also is deleting the recents from the send flow and it's underneath logic

**Technical detais**
AddressElement, AddressList, and IdentIcon updated to typescript and to functional components.

Deleted recents from the redux state, freeing more memory. Created a migration for the existing users.

**Screenshots/Recordings**
http://recordit.co/7eR0wBAfbA

**E2E build**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/c3e17674-c28d-4790-a402-1eadaf3fb1ad
**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by tommasini** [comment]
Regarding the SonarCloud bug: 
image
The intention it's to order alphabetically so now need to address it on my opinion, but let me know your thoughts on this

File: app/components/Views/SendFlow/AddressElement/index.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   * Callback on account press
+   */
+  onAccountPress: (address: string) => void;
+  /**
+   * Callback on account long press
+   */
+  onAccountLongPress: (address: string) => void;
+}
+
+const createStyles = (colors: any) =>
```
Comment (sethkfman) [inline app/components/Views/SendFlow/AddressElement/index.tsx:37]: Update these TypeScript files with the recommended file structure (e.g. breakout styles, types and component)

Comment (tommasini) [reply app/components/Views/SendFlow/AddressElement/index.tsx:37]: Thanks for the review Seth!! Will do it!!

**Comment by tommasini** [comment]
Rebuilt entirely pipeline with the latest changes: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/c3e17674-c28d-4790-a402-1eadaf3fb1ad

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**